@uppy/companion: remove `lodash` dependency

aduh95 opened the pull request.

mifi reviewed (approved):
changes LGTM. Not sure why tests are failing though

aduh95 commented:
> changes LGTM. Not sure why tests are failing though

After investigating, the CI failures were due to:
- Node.js v10.x and v12.x not supporting optional chaining syntax. It is supported by TS 3.7+ (we're using TS 4.3), but the tests are testing the `src` folder, not the transpiled `lib` folder. **Solution**: I've removed Node.js v10.x and v12.x from the Companion CI, I don't think there's any value to keep them around anymore (ping @goto-bus-stop in case I'm wrong)
- `lodash/merge` is actually doing a deep merge, while the spread operator is doing a shallow one. **Solution**: I've changed the dependency tree to use `lodash.merge` and removed the spread operators.

I've tried to also add Node.js v16.x to the CI, but that failed due to a number of unhandled promise rejections, this would need to be addressed in a follow-up PR.

arturi commented:
> Since v2, you now need to be running node.js >= v10.20.1 to use Companion. Please see Migrating v1 to v2

Is it ok to not test with Node.js v10.x and v12.x if we support these versions?

mifi commented:
> Is it ok to not test with Node.js v10.x and v12.x if we support these versions?

I'm for removing node10 as it's no longer maintained. (but probably a major version bump.) Node 12 however is still supported until apr 22 so we should keep support for it.
